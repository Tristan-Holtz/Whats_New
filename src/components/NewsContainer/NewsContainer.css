.news-section {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 25px;
  margin: 2%;
  overflow: auto;
  width: 100%;
}
