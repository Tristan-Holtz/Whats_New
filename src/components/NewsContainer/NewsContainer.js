import React from 'react';
import './NewsContainer.css';
import NewsArticle from '../NewsArticle/NewsArticle.js';

const NewsContainer = props => {
  return (
    <section className="news-section">
      {props.currentPage.map(article => (
        <NewsArticle
          url={article.url}
          headline={article.headline}
          img={article.img}
          description={article.description}
        />
      ))}
    </section>
  );
};

export default NewsContainer;
