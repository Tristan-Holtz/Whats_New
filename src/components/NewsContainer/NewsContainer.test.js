import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  it('renders a container of articles', () => {
    const wrapper = shallow(
      <NewsContainer
        currentPage={[
          {
            id: 1,
            headline: 'The Who postpones Denver Concert at the Pepsi Center',
            img:
              'https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg',
            description:
              'Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.',
            url:
              'https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/'
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
