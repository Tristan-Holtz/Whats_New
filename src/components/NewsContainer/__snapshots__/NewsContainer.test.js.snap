// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer renders a container of articles 1`] = `ShallowWrapper {}`;
